lib/mixins/flowdock.ts

```diff
@@ -0,0 +1,21 @@
+import * as request from 'request-promise';
+import { FlowdockInboxItem } from './flowdock-types';
+
+export class FlowdockAdapter {
+    public postToInbox(item: FlowdockInboxItem) {
```

Comment:
One field with `public X = new X()` per service feels simpler than `implements X` + `applyMixin` + the stub function field, per service, especially at runtime. I've seen mixins used when you want to pull in a set of functions to add to your external api, and that's a great use case, but I don't think they works well if you only need to use them internally like this.

I am all for future-proofing and a declarative approach ofc. If do we want to really flexibly add lots of other services though, you want a way to set up the config for each service up front (in a perfect world, with one shared config), and then have standardized postError/postSuccess functions you can call for all of them at once, so the `reportError` code is simpler and notifier-agnostic. An instance-based example:

```typescript
class VersionBot {
  constructor(notifierConstructors: NotifierConstructor[]) {
    // Might not be practical to do all config upfront like this, but if not you can
    // always do per-service setup outside and pass in notifier instances.
    this.notifiers = notifierConstructors.map((Notifier) => new Notifier({
      tags: [ 'devops' ],
      from_address: 'versionbot@resin.io',
      ...
    }));
  }

  reportError(error: Error) {
    this.notifiers.forEach((notifier) => notifier.postError(error))
  }
}

export = new VersionBot([
  FlowdockNotifier,
  EmailNotifier,
  ... 
]);
```

With mixins you can't do this. Each mixin adds at least one separate function to the class, and you can't loop over them, so you can't decouple yourself from the functions involved, and you can't do any setup before the postToInbox/postError/whatever call either, so you have to configure them individually at notify time, unless you bring in MixinFactories (ick).

Not saying we should jump to the above until this is a real problem, just that mixins won't scale as a nice solution for this kind of thing. Simple objects stay way more flexible.